src/ng/httpBackend.js

```diff
@@ -1,7 +1,33 @@
 'use strict';
 
-function createXhr() {
-    return new window.XMLHttpRequest();
+/**
+ * @ngdoc service
+ * @name $xhrFactory
+ *
+ * @description
+ * Factory function used to create XMLHttpRequest objects.
+ *
+ * Decorate this service to create your own custom XMLHttpRequest objects.
+ *
+ * ```
+ * .config(function($provide) {
+ *   $provide.decorator('$xhrFactory', function() {
```

Comment:
This example is not ideal as here we are replacing the entire implementation really. I would simply show how to register a factory with the same (`$xhrFactory`, that is) name.